client/components/token-field/index.jsx

```diff
@@ -181,6 +183,10 @@ var TokenField = React.createClass( {
 			onBlur: this._onBlur,
 		};
 
+		if ( value.length < 1 && placeholder ) {
+			props = { ...props, placeholder };
```

Comment:
I know there's precedent, but I think we should just mutate the prop object directly since we've created it within the scope of this function. It saves unnecessary [property cloning](https://github.com/zloirock/core-js/blob/master/modules/_object-assign.js) for an object we shouldn't care to clone. Instead:

```js
props.placeholder = placeholder;
```